packages/xstate-wallet/src/workflows/confirm-create-channel.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  MachineConfig,
+  Action,
+  DefaultGuardType,
+  GuardPredicate,
+  ConditionPredicate,
+  StateSchema,
+  Machine
+} from 'xstate';
+import {Allocations, Participant} from '@statechannels/client-api-schema';
+import {MachineFactory, Store} from '@statechannels/wallet-protocols';
+import {sendDisplayMessage} from '../messaging';
+
+interface WorkflowActions {
+  hideUi: Action<WorkflowContext, any>;
+  displayUi: Action<WorkflowContext, any>;
+}
+interface WorkflowGuards {
+  noBudget:
+    | GuardPredicate<WorkflowContext, WorkflowEvent>
+    | ConditionPredicate<WorkflowContext, WorkflowEvent>;
+}
+// While this context info may not be used by the workflow
+// it may be used when displaying a UI
+interface WorkflowContext {
+  participants: Participant[];
+  allocations: Allocations;
+  appDefinition: string;
+  appData: string;
+  chainId: string;
+  challengeDuration: number;
```

Comment:
Probably need to add the `requestId` here, to allow the JSON-RPC interface to include it in the result.

Reply:
Will the `requestId` live that long? When a user calls `create-channel` I'm assuming we will return success immediately instead of blocking until a user clicks something (the current `channel-client` will timeout and retry if we don't respond fairly quickly)

 Further updates will be provided to the client over json-rpc notifications.

Reply:
As it currently works, that call returns a `channel-result`, which I think would mean that we need to hang onto the `requestId` until we have the channel created. We could change that though.

Reply:
Ah right. We probably don't want to to block the json-rpc call so we should probably change that.